ansible_builder/main.py

```diff
@@ -329,26 +329,46 @@ def prepare_galaxy_install_steps(self):
             self.steps.extend(GalaxyInstallSteps(CONTEXT_FILES['galaxy']))
         return self.steps
 
-    def prepare_pip_steps(self, pip_lines):
-        if ''.join(pip_lines).strip():  # only use file if it is non-blank
-            pip_file = os.path.join(self.build_context, PIP_COMBINED)
-            write_file(pip_file, pip_lines)
-            self.steps.extend(PipSteps(PIP_COMBINED))
+    def prepare_assemble_steps(self):
+        requirements_file_exists = os.path.exists(os.path.join(self.build_context, PIP_COMBINED))
+        if requirements_file_exists:
+            self.steps.append("ADD {0} /tmp/src/requirements.txt".format(PIP_COMBINED))
+
+        bindep_exists = os.path.exists(os.path.join(self.build_context, BINDEP_COMBINED))
+        if bindep_exists:
+            self.steps.append("ADD {0} /tmp/src/bindep.txt".format(BINDEP_COMBINED))
+
+        if requirements_file_exists or bindep_exists:
+            self.steps.append("RUN assemble")
```

Comment:
If we were being absolutists about code style, every string that goes in the Dockerfile would originate in the `ansible_builder.steps` module. If not, I envision the steps module eventually going away.

Reply:
I might further refactor this later. Dunno yet.